src/bridge/cockpitdbusjson.c

```diff
@@ -476,37 +500,38 @@ parse_json (JsonNode *node,
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_BYTE))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "uint8", 0, G_MAXUINT8, error))
             return g_variant_new_byte (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_INT16))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "int16", G_MININT16, G_MAXINT16, error))
             return g_variant_new_int16 (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_UINT16))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "uint16", 0, G_MAXUINT16, error))
             return g_variant_new_uint16 (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_INT32))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "int32", G_MININT32, G_MAXINT32, error))
             return g_variant_new_int32 (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_UINT32))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "uint32", 0, G_MAXUINT32, error))
             return g_variant_new_uint32 (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_INT64))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          if (check_int_type (node, "int64", G_MININT64, G_MAXINT64, error))
             return g_variant_new_int64 (json_node_get_int (node));
         }
       else if (g_variant_type_equal (type, G_VARIANT_TYPE_UINT64))
         {
-          if (check_type (node, JSON_NODE_VALUE, G_TYPE_INT64, error))
+          /* can never be larger than signed int64 */
+          if (check_int_type (node, "uint64", 0, G_MAXINT64, error))
```

Comment:
This maintains current behaviour, but the comment doesn't really explain anything. It should say *why* this can't be bigger than `G_MAXINT64`. glib, D-Bus etc. certainly don't have this limitation, and JavaScript is completely odd (everything is a double, so with large numbers you'll lose precision but they are not bounded) so where does it come from?

My guess is that it merely comes from the signature and types in `check_int_type()`, as that by itself cannot accept guint64. I'd rather recommend using a separate `check_guint64_type()` for that, this seems more robust than arbitrarily breaking an official D-Bus data type.